File: farebot-app-persist/src/main/java/com/codebutler/farebot/persist/CardKeysPersister.java
package com.codebutler.farebot.persist;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.codebutler.farebot.persist.db.model.SavedKey;

import java.util.List;

public interface CardKeysPersister {

    @NonNull
    List<SavedKey> getSavedKeys();

    @Nullable
    SavedKey getForTagId(@NonNull String tagId);

    long insert(@NonNull SavedKey savedKey);

    void delete(@NonNull SavedKey savedKey);
}

File: app/src/commonMain/kotlin/com/codebutler/farebot/persist/CardKeysPersister.kt
package com.codebutler.farebot.persist

import com.codebutler.farebot.persist.db.model.SavedKey

interface CardKeysPersister {
    fun getSavedKeys(): List<SavedKey>

    fun getForTagId(tagId: String): SavedKey?

    fun insert(savedKey: SavedKey): Long

    fun delete(savedKey: SavedKey)

    fun getGlobalKeys(): List<ByteArray>

    fun insertGlobalKeys(keys: List<ByteArray>, source: String)

    fun deleteAllGlobalKeys()
}
